gateway-ha/src/main/java/io/trino/gateway/ha/router/StatementUtils.java

```diff
@@ -191,13 +193,17 @@ private StatementUtils() {}
             .build().stream()
             .collect(toImmutableMap(StatementTypeInfo::getStatementType, identity()));
 
-    public static Optional<QueryType> getQueryType(Statement statement)
+    public static String getResourceGroupQueryType(Statement statement)
     {
         if (statement instanceof ExplainAnalyze) {
-            return getQueryType(((ExplainAnalyze) statement).getStatement());
+            return getResourceGroupQueryType(((ExplainAnalyze) statement).getStatement());
+        }
+        StatementTypeInfo<? extends Statement> statementTypeInfo = STATEMENT_QUERY_TYPES.get(statement.getClass());
+        if (statementTypeInfo != null) {
+            return statementTypeInfo.getQueryType().toString();
         }
-        return Optional.ofNullable(STATEMENT_QUERY_TYPES.get(statement.getClass()))
-                .map(StatementTypeInfo::getQueryType);
+        log.warn("Encountered unknown statement type: %s", statement.getClass());
```

Comment:
We usually log messages when encountered something. I would remove "Encountered" from the message. 
```suggestion
        log.warn("Unsupported statement type: %s", statement.getClass());
```